File: Assignment38_1.c
// Write a program which accepts string from user and copy characters of one string to another string in reverse order

#include<stdio.h>

void strcpyX(char *src, char *dest)
{
    int ilength = 0;

    while(*src != '\0')
    {
        src++;
        ilength++;
    }
    src--;

    while(ilength > 0)
    {
        *dest = *src;
        src--;
        dest++;
        ilength--;
    }
    *dest = '\0';
}

int main()
{
    char Arr[100] = {'\0'};
    char Brr[100] = {'\0'};

    printf("Enter a String: \n");
    scanf("%[^'\n']s",Arr);

    strcpyX(Arr, Brr);
    
    printf("Updated string is :%s\n",Brr);

    return 0;
}

File: Assignment38_1.java
// Write a program which accepts string from user and copy characters of one string to another string in reverse order
import java.util.*;
class Logic
{
    public void StrcpyX(String str)
    {
        char Arr[] = str.toCharArray();
        char Brr[] = new char[Arr.length];
        int i = 0, j = 0;
        

        for(i = Arr.length-1; i >=0; i--, j++)
        {
            Brr[j] = Arr[i];                      
        }

        System.out.println("Updated string is: "+new String(Brr));
    }
}


class Assignment38_1
{
    public static void main(String A[])
    {
        String sobj = null;
        Scanner scanobj = new Scanner(System.in);

        System.out.println("Enter a string: ");
        sobj = scanobj.nextLine();


        Logic lobj = new Logic();
        lobj.StrcpyX(sobj);
        
    }
}